control-plane/connect-inject/common/common.go

```diff
@@ -192,6 +192,16 @@ func HasBeenMeshInjected(pod corev1.Pod) bool {
 	return false
 }
 
+func IsGateway(pod corev1.Pod) bool {
+	if pod.Annotations == nil {
+		return false
+	}
+	if anno, ok := pod.Annotations[constants.AnnotationGatewayKind]; ok && anno != "" {
```

Comment:
This utilizes a known annotation added to all v2 mesh gateway (and api/terminating in the future) pods

https://github.com/hashicorp/consul-k8s/blob/c8e5d9e27e78f021630d708b88a3387eeec90ab5/control-plane/gateways/deployment.go#L65